sdk/JavaScript/TestFramework/js/tests/pushTests.js

```diff
@@ -76,298 +77,187 @@ function definePushTestsNamespace() {
-            '<text id="4">tl-spiat1-4</text>' +
-            '</binding></visual></tile>'));
-    tests.push(createPushTest('sendTileSquareBlock',
-    { text1: '24', text2: 'aliquam' },
-    '<tile><visual><binding template="TileSquareBlock"><text id="1">24</text><text id="2">aliquam</text></binding></visual></tile>'));
-
-    tests.push(new zumo.Test('Unregister push channel', function (test, done) {
-        if (pushChannel) {
-            var runtimeFeatures = zumo.util.globalTestParams[zumo.constants.RUNTIME_FEATURES_KEY];
-            if (runtimeFeatures[zumo.runtimeFeatureNames.NH_PUSH_ENABLED]) {
-                zumo.getClient().push.unregisterNative().done(function () {
-                    test.addLog('Unregistered with NH: ');
-                    done(true);
-                }, function (error) {
-                    test.addLog('Failed to unregister with NH: ', error);
-                    done(false);
-                });
-            }
-            else {
-                pushChannel.close();
-                done(true);
-            }
-        } else {
-            test.addLog('Error, push channel needs to be registered.');
-            done(false);
-        }
-        pushChannel = null;
+    tests.push(new zumo.Test('UnregisterTemplate', function (test, done) {
+        testPlatform.getPushChannel()
+            .then(function (channel) {
+                channelUri = channel.uri;
+                return zumo.getClient().push.unregisterTemplate(testTemplate.name);
+            })
+            .then(getRegistrations(channelUri, 'wns'))
+            .then(function (registrations) {
+                zumo.assert.isTrue(Array.isArray(registrations));
+                zumo.assert.areEqual(0, registrations.length);
+                zumo.assert.areEqual('{}', JSON.stringify(zumo.getClient().push._registrationManager._storageManager._registrations));
+                return true;
+            })
+            .done(done, fail(test, done));
+    }));
+    
+    tests.push(new zumo.Test('RegisterMultiple', function (test, done) {
+        testPlatform.getPushChannel()
+            .then(function (channel) {
+                channelUri = channel.uri;
+                return zumo.getClient().push.registerNative(channelUri);
+            })
+            .then(function () {
+                return zumo.getClient().push.registerTemplate(channelUri, testTemplate.body, testTemplate.name, testTemplate.headers, testTemplate.tags);
+            })
+            .then(function () {
+                return zumo.getClient().push.registerNative(channelUri);
+            })
+            .then(wait(registrationWaitInterval))
+            .then(getRegistrations(channelUri, 'wns'))
+            .then(function (registrations) {
+                zumo.assert.isTrue(Array.isArray(registrations));
+                zumo.assert.areEqual(2, registrations.length);
+                return true;
+            })
+            .then(function () {
```

Comment:
nit: The above three lines of code are not required - simply return the promise from unregisterAll in the same block.